spring-boot/admin/src/main/java/eu/coatrack/admin/service/GatewayHealthMonitorService.java

```diff
@@ -0,0 +1,137 @@
+package eu.coatrack.admin.service;
+
+/*-
+ * #%L
+ * coatrack-admin
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import eu.coatrack.admin.model.repository.ProxyRepository;
+import eu.coatrack.api.Proxy;
+import eu.coatrack.api.ProxyStates;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/*
+ * This class calculates and provides all data required for the Gateway Health Monitor.
+ */
+@Service
+public class GatewayHealthMonitorService {
+
+    @Value("${ygg.gateway-health-monitor.warning.threshold.minutes}")
+    private int gatewayHealthWarningThresholdInMinutes;
```

Comment:
This is field injection which is not recommended. As this dependency is required to make this class work I suggest to initialize it via constructor injection.

Reply:
I remember you mentioned this before, but so that I am curious and also want to learn.
Why is the field injection not recommended?

Reply:
Here is a good explanation: https://stackoverflow.com/questions/39890849/what-exactly-is-field-injection-and-how-to-avoid-it

The drawback in your case was that you were forced to use reflection (which should be avoided) to test this class because the variable was not accessible via constructor or setter.

Reply:
Done, and lesson learned ;D
From now on, I will use constructors instead of field dependency injection